src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -75,6 +78,9 @@
  * automatically public, static and final just as their
  * annotation fields are automatically public and abstract.</p>
  *
+ * <p>Nested records are implicitly static. This avoids an immediately enclosing
+ * instance which would silently add state to the record class.</p>
```

Comment:
please add some link to jdk spec (or something similar) to make it clearer for users

Reply:
Done.